packages/core/src/components/ModalNew/Modal/Modal.tsx

```diff
@@ -67,41 +74,54 @@ const Modal = forwardRef(
       keys: [keyCodes.ESCAPE]
     });
 
-    if (!show) {
-      return null;
-    }
+    const modalAnimationVariants = anchorElementRef?.current
+      ? modalAnimationAnchorPopVariants
+      : modalAnimationCenterPopVariants;
 
     return (
-      <ModalProvider value={contextValue}>
-        <RemoveScroll>
-          <div
-            data-testid={getTestId(ComponentDefaultTestId.MODAL_NEXT_OVERLAY, id)}
-            className={styles.overlay}
-            onClick={onBackdropClick}
-            aria-hidden
-          />
-          <FocusLock returnFocus>
-            <div
-              ref={ref}
-              className={cx(styles.modal, getStyle(styles, camelCase("size-" + size)), className)}
-              id={id}
-              data-testid={dataTestId || getTestId(ComponentDefaultTestId.MODAL_NEXT, id)}
-              role="dialog"
-              aria-modal
-              aria-labelledby={titleId}
-              aria-describedby={descriptionId}
-            >
-              <ModalTopActions
-                renderAction={renderHeaderAction}
-                color={closeButtonTheme}
-                closeButtonAriaLabel={closeButtonAriaLabel}
-                onClose={onClose}
+      <AnimatePresence>
+        {show && (
+          <ModalProvider value={contextValue}>
+            <RemoveScroll>
+              <motion.div
+                variants={modalAnimationOverlayVariants}
+                initial="exit"
+                animate="enter"
+                exit="exit"
+                data-testid={getTestId(ComponentDefaultTestId.MODAL_NEXT_OVERLAY, id)}
+                className={styles.overlay}
+                onClick={onBackdropClick}
+                aria-hidden
               />
-              {children}
-            </div>
-          </FocusLock>
-        </RemoveScroll>
-      </ModalProvider>
+              <FocusLock returnFocus>
+                <motion.div
+                  variants={modalAnimationVariants}
+                  initial="exit"
+                  animate="enter"
+                  exit="exit"
+                  custom={anchorElementRef}
```

Comment:
Does this need a fallback in case it is undefined, so it won't mess up the animation?

Reply:
checked it, if it is undefined then the pop from center variation would be applied and it doesn't use it
if it is being used from `custom`, then the pop from anchor variation would be applied, and it means that the anchor element ref isn't undefined